hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/file/FileTableFunction.java

```diff
@@ -83,6 +95,52 @@ private static String randomName() {
         return SCHEMA_NAME_FILES + "_" + UuidUtil.newUnsecureUuidString().replace('-', '_');
     }
 
+    @Override
+    public List<Permission> permissions(SqlCall call) {
+        // Try find operand by name first, most common option.
+        SqlNode astPath = findOperandByName(OPTION_PATH, call);
+        // If not successful -> try to find by position: SELECT * FROM TABLE(fn('path', 'glob'))
+        if (astPath == null) {
+            astPath = findOperandByPosition(0, call);
+        }
+        if (astPath == null) {
+            throw QueryException.error("Can't determine the file path, query is denied in secure environment");
+        }
+        if (astPath instanceof SqlLiteral) {
+            String path = astPath.toSqlString(PostgresqlSqlDialect.DEFAULT).getSql();
+            // Strip quotes
+            if (path.charAt(0) == '\'' && path.charAt(path.length() - 1) == '\'') {
```

Comment:
this implementation is inconsistent with `HazelcastDynamicTableFunction.extractStringValue` which also seems simpler

Reply:
afeb397c9812fee94a0cd4239d2184f4660d997a